### PR TITLE
[16.0][MIG] mail_tracking: Migration to 16.0

### PR DESCRIPTION
**Pull request by maq-adhoc**
I restart the mail tracking migration, taking changes from PR https://github.com/OCA/social/pull/1029 and adding my own changes. 
I restarted to be able to do it with the odoo-module-migrate tool and the commits are clean and include PRs #1053 ,#1056 ,#1057.  
I also modified the way we inherit MessagingInitializer so that it does not conflict with other modules, e.g. Voip.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
There's already #1029, so please review that one. Closing for now.

**Comment by pedrobaeza** [comment]
Sorry, I have seen the comments, so this one may be the correct one. Please check pre-commit.

File: mail_tracking/static/src/xml/failed_message/discuss.xml
Comment (remi-filament) [inline mail_tracking/static/src/xml/failed_message/discuss.xml]:
```suggestion
            <t t-elif="thread.mailbox === messaging.failedmsg">
                <div class="o_ThreadIcon_mailboxModeration fa fa-fw fa-exclamation" />
```
The failedmsg mailbox is showing without icon without the above change

File: mail_tracking/static/src/xml/failed_message/discuss.xml
Comment (remi-filament) [inline mail_tracking/static/src/xml/failed_message/discuss.xml]:
```suggestion
                Failed messages will be listed here.
```

File: mail_tracking/readme/USAGE.rst
Comment (remi-filament) [inline mail_tracking/readme/USAGE.rst:64]: Image is quite old and could be replaced by a more up to date UI capture

**Review by remi-filament — CHANGES_REQUESTED** [review]
Hi @maq-adhoc, Thank you for this port !
I am not sure how to test this one extensively, so far I have spotted a problem of display on Discuss page.

Some other issues I found (things that I have verified are working in v15) : 
-  I do not see any message in Failedmsg on Discuss app, although I guess there should be the demo message created for that purpose no ?
- Failed sent messages filter is not added in search views
- Action in Discuss App to "Set as reviewed" or "Retry" have no effect

File: mail_tracking/readme/USAGE.rst
Comment (maq-adhoc) [reply mail_tracking/readme/USAGE.rst:64]: thanks @remi-filament the errors in Failedmsg had been detected and I am already working to fix them. I take note of the rest of the comments to verify and correct them.

File: mail_tracking/models/mail_tracking_email.py
Comment (kmagusiak) [inline mail_tracking/models/mail_tracking_email.py]:
```suggestion
        string="Message (mail)", comodel_name="mail.message", readonly=True, index=True
```
Name conflicts with `message_id` field.

File: mail_tracking/models/ir_mail_server.py
Comment (nguyenminhchien) [inline mail_tracking/models/ir_mail_server.py]: @maq-adhoc it should be `smtp_ssl_private_key=smtp_ssl_private_key` instead of `smtp_ssl_private_key=smtp_ssl_certificate`

File: mail_tracking/models/mail_thread.py
Comment (nguyenminhchien) [inline mail_tracking/models/mail_thread.py]: @maq-adhoc  Method `_fields_view_get` is deprecated, use `_get_view` instead

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]
I added 2 comments, please check them. Thank you.

**Comment by TDu** [comment]
@maq-adhoc  Could you include this fix https://github.com/OCA/social/pull/1139 that has been merged into 14. Thank you

**Comment by pedrobaeza** [comment]
Please cherry-pick #1141

**Comment by pedrobaeza** [comment]
Please cherry-pick #1149

File: mail_tracking/models/mail_thread.py
Comment (nguyenminhchien) [inline mail_tracking/models/mail_thread.py]: I face an error when accessing the Sales menu. Seems that you don't test before pushing your changes.
image

Comment (maq-adhoc) [reply mail_tracking/models/mail_thread.py]: It was a work commit. I fixed that error. The module is functional for almost everything.

Not all the problems are solved, e.g. it doesn't list all the error cases in the discuss/failed msg menu. This week I think the final version will be ready.

File: mail_tracking/readme/USAGE.rst
Comment (nguyenminhchien) [reply mail_tracking/readme/USAGE.rst:64]: @maq-adhoc i just created this [Pull request](https://github.com/adhoc-dev/social/pull/1) to fix:
- Failedmsg on Discuss app
- Action in Discuss App to "Set as reviewed" or "Retry" have no effect

Please review
Thanks,

**Comment by maq-adhoc** [comment]
> I added 2 comments, please check them. Thank you.
@nguyenminhchien 

They are made in the current commit. 

https://github.com/OCA/social/pull/1091/files#diff-3bb501b5e868a62e76b5c295015ee18984cf388955d4b160442f145b39c3e694R121
https://github.com/OCA/social/pull/1091/files#diff-cbb61b601dc8e9b4a1de5d64eeabd4f63cffa4f007f7cad220787ea714a3f83eR90

**Comment by nguyenminhchien** [comment]
> They are made in the current commit.

@maq-adhoc thanks. But, i just found an issue need to be migrated: Showing the Failedmsg on the Chatter
image

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]
**Failed msg** on the Chatter is not shown

**Comment by nguyenminhchien** [comment]
@maq-adhoc : i created this [PR](https://github.com/adhoc-dev/social/pull/2) to show the Failed Messages on Chatbox

**Review by nguyenminhchien — APPROVED** [review]
lgtm

**Comment by patel1008b** [comment]
https://github.com/OCA/social/pull/1191

File: mail_tracking/static/src/js/models/mailbox.esm.js
Comment (payen000) [inline mail_tracking/static/src/js/models/mailbox.esm.js:26]:
```suggestion
                        return this.env._t("Failed");
```

To maintain consistency with previous versions, I think this shoud say "Failed" instead.

File: mail_tracking/static/src/js/failed_message/thread.esm.js
Comment (payen000) [inline mail_tracking/static/src/js/failed_message/thread.esm.js:1]: This file seems to not be needed anymore, so we can delete it

File: mail_tracking/static/src/js/failed_message/discuss.js
Comment (payen000) [inline mail_tracking/static/src/js/failed_message/discuss.js:1]: Is this file in use? If not, I think we should delete it before merging

**Review by payen000 — COMMENTED** [review]
I tested this functionally and seems to be working as expected.

This will be ready to merge once the unused files are deleted and the commits are sorted out (as there are too many commits right now).

Edit:

My bad, I didn't realize the many commits were the module's commit history.

**Comment by payen000** [comment]
Hello @maq-adhoc,

I left some comments for you, could you check them please?

I believe the module is working as it should and will be ready to merge once we delete the unused file(s) and switch "Failedmsg" by "Failed", or is there anything else that I'm missing?

**Review by cvinh — CHANGES_REQUESTED** [review]
Hello
Could you please forward port this PR : https://github.com/OCA/social/pull/941 ?
Thanks

**Comment by payen000** [comment]
Superseded by https://github.com/OCA/social/pull/1216

However, I do believe this PR was already working properly, I'm only opening the other PR due to inactivity here, as I need the module working in `v16` as soon as possible.

**Pull request closed without merging**
